### ISSUE TITLE
does not compile on windows

### ISSUE THREAD

**Issue by csarn** [SunHao-0/libtcc#1]
I'm getting this error when trying to use libtcc on windows 10:

    error: failed to run custom build command for `libtcc v0.2.0`

    Caused by:
      process didn't exit successfully: `target\debug\build\libtcc-a6841ebbe154479f\build-script-build` (exit code: 1)
    warning: build failed, waiting for other jobs to finish...
    error: build failed

**Comment by SunHao-0** [comment]
I need more information to solve this issue.
Please show me your stdout or stderr logs during the building process which should be available in your [Project]/target/[debug/release]/build/libtcc-* path.

**Comment by SunHao-0** [comment]
Normally, you need  MinGW and MSYS tools to build *tcc* locally(more info: https://github.com/TinyCC/tinycc/blob/mob/win32/tcc-win32.txt#L109), then build libtcc. 
Currently, this crate only builds tcc from source in the cross-compile situation for correctness.

**Comment by csarn** [comment]
I figured out that `cargo build -vv` also displays the build.rs output.
Here it is:

    [libtcc 0.2.0] OPT_LEVEL = Some("1")
    [libtcc 0.2.0] TARGET = Some("x86_64-pc-windows-msvc")
    [libtcc 0.2.0] HOST = Some("x86_64-pc-windows-msvc")
    [libtcc 0.2.0] CC_x86_64-pc-windows-msvc = None
    [libtcc 0.2.0] CC_x86_64_pc_windows_msvc = None
    [libtcc 0.2.0] HOST_CC = None
    [libtcc 0.2.0] CC = None
    [libtcc 0.2.0] CFLAGS_x86_64-pc-windows-msvc = None
    [libtcc 0.2.0] CFLAGS_x86_64_pc_windows_msvc = None
    [libtcc 0.2.0] HOST_CFLAGS = None
    [libtcc 0.2.0] CFLAGS = None
    [libtcc 0.2.0] CRATE_CC_NO_DEFAULTS = None
    [libtcc 0.2.0] CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
    [libtcc 0.2.0] DEBUG = Some("true")
    [libtcc 0.2.0] running "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional\\VC\\Tools\\MSVC\\14.10.25017\\bin\\HostX64\\x64\\cl.exe" "src/libtcc_test.c" "-o" "C:\\Users\\SARNOW4\\Home\\Code\\Rust\\analyzer\\target\\debug\\build\\libtcc-277853c1fdc58435\\out\\tcc-tmp\\a.out" "-Isrc/tcc-0.9.27" "-ltcc" "-ldl" "-lrt" "-lm" "-lpthread"
    [libtcc 0.2.0] Microsoft (R) C/C++-Optimierungscompiler Version 19.10.25019 f�r x64
    [libtcc 0.2.0] Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.
    [libtcc 0.2.0]
    [libtcc 0.2.0] cl : Befehlszeile warning D9035 : Die Option "o" ist veraltet und wird in einer der n�chsten Versionen entfernt.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-ltcc" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-ldl" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-lrt" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-lm" wird ignoriert.
    [libtcc 0.2.0] cl : Befehlszeile warning D9002 : Unbekannte Option "-lpthread" wird ignoriert.
    [libtcc 0.2.0] libtcc_test.c
    [libtcc 0.2.0] src/libtcc_test.c(4): fatal error C1083: Datei (Include) kann nicht ge�ffnet werden: "stdlib.h": No such file or directory
    [libtcc 0.2.0] ERROR: Can not find libtcc.a in your host:
    [libtcc 0.2.0]  Tcc should be installed in host when your build target is same as host,
    [libtcc 0.2.0]  because libtcc need some small but necessary runtime libaray such as libtcc1.a
    [libtcc 0.2.0]  and some header files, which should be found in [prefix]/lib/tcc
    error: failed to run custom build command for `libtcc v0.2.0`

The trouble seems to be:
1. if cc finds the MSVC compiler, all the compiler arguments hardcoded in build.rs are wrong
2. the "[prefix]/lib/tcc" convention is for unix, not windows. So as a windows user, I have no idea where to put everything from tinycc so that the build script is able to pick it up

**Comment by SunHao-0** [comment]
> 1. if cc finds the MSVC compiler, all the compiler arguments hardcoded in build.rs are wrong

I've made some modifications to build.rs. For now, build.rs won't check whether tcc is installed or not on windows. Instead, it just tells cargo to link native tcc lib. Therefore, users should make sure tcc is installed correctly.
   
> 2. the "[prefix]/lib/tcc" convention is for unix, not windows. So as a windows user, I have no idea where to put everything from tinycc so that the build script is able to pick it up

You may use MinGW and MSYS tools to build tcc locally and use *LIB_TCC* environment variable to specify the linking path. 


Checkout [https://github.com/SunHao-0/libtcc/tree/fix-%231](url) branch for more information.
Feel free to send a PR if you don't satisfy with this sample solution*_*.

**Comment by csarn** [comment]
Ok, thanks for the quick fix! With minor modification I got it working!

Here is what I did:

In a Visual Studio Command Prompt, change to src\tcc-0.9.27\win32 and run `build-tcc.bat -c cl`
Inside the root of this repository's checkout (fix-#1 branch):
- set `LIB_TCC` to src\tcc-0.9.27\win32
- add that directory to `PATH`
- and


change

    println!("cargo:rustc-link-lib=static=tcc");

to

    println!("cargo:rustc-link-lib=static=libtcc");

because the msvc linker does not add the `lib` prefix automatically

Then a `cargo run --example greet` worked :)

What made the debugging really hard is that cargo seems to cache the results from build.rs, so after I changed my LIB_TCC environment variable, I also had to `touch build.rs` so that cargo would run it again and pick up the env changes.
